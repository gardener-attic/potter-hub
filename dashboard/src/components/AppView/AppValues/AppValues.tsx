import * as React from "react";
import AceEditor from "react-ace";

interface IAppValuesProps {
  values: string;
}

const AppValues: React.SFC<IAppValuesProps> = props => {
  if (props.values === "") {
    return (
      <React.Fragment>
        <h6>Installation Values</h6>
        <p>The current application was installed without specifying any values</p>
      </React.Fragment>
    );
  }
  return (
    <React.Fragment>
      <h6>Installation Values</h6>
      <AceEditor
        mode="yaml"
        theme="xcode"
        name="values"
        width="100%"
        maxLines={40}
        setOptions={{ showPrintMargin: false }}
        editorProps={{ $blockScrolling: Infinity }}
        value={props.values}
        readOnly={true}
      />
    </React.Fragment>
  );
};

export default AppValues;
