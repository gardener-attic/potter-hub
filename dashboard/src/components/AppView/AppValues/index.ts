import AppValues from "./AppValues";

export default AppValues;
