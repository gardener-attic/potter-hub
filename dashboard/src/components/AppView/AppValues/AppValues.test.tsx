import { shallow } from "enzyme";
import * as React from "react";
import AppValues from "./";

it("match snapshot with values", () => {
  const wrapper = shallow(<AppValues values="foo: bar" />);
  expect(wrapper).toMatchSnapshot();
});

it("match snapshot without values", () => {
  const wrapper = shallow(<AppValues values="" />);
  expect(wrapper).toMatchSnapshot();
});
