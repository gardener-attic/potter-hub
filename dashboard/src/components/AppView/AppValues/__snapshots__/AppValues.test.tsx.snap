// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot with values 1`] = `
<Fragment>
  <h6>
    Installation Values
  </h6>
  <ReactAce
    cursorStart={1}
    editorProps={
      Object {
        "$blockScrolling": Infinity,
      }
    }
    enableBasicAutocompletion={false}
    enableLiveAutocompletion={false}
    enableSnippets={false}
    focus={false}
    fontSize={12}
    height="500px"
    highlightActiveLine={true}
    maxLines={40}
    minLines={null}
    mode="yaml"
    name="values"
    navigateToFileEnd={true}
    onChange={null}
    onLoad={null}
    onPaste={null}
    onScroll={null}
    placeholder={null}
    readOnly={true}
    scrollMargin={
      Array [
        0,
        0,
        0,
        0,
      ]
    }
    setOptions={
      Object {
        "showPrintMargin": false,
      }
    }
    showGutter={true}
    showPrintMargin={true}
    style={Object {}}
    tabSize={4}
    theme="xcode"
    value="foo: bar"
    width="100%"
    wrapEnabled={false}
  />
</Fragment>
`;

exports[`match snapshot without values 1`] = `
<Fragment>
  <h6>
    Installation Values
  </h6>
  <p>
    The current application was installed without specifying any values
  </p>
</Fragment>
`;
